src/finchlite/finch_assembly/__init__.py

```diff
@@ -1,5 +1,11 @@
+from .assembly_dataflow import (
```

Comment:
Could we keep cfg_builder in Finch assembly, but move the dataflow passes to a separate folder outside of finch assembly? I think of the specific dataflow analyses as something separate from the definition of the IR.